var/spack/repos/builtin/packages/py-pybind11/package.py

```diff
@@ -64,3 +66,8 @@ def cmake_args(self):
                 'ON' if self.run_tests else 'OFF')
         ]
         return args
+
+    def install(self, spec, prefix):
+        super(PyPybind11, self).install(spec, prefix)
+        os.environ['PYBIND11_USE_CMAKE'] = '1'
```

Comment:
there is a `setup_environment` function that you can override, see [here](https://spack.readthedocs.io/en/latest/tutorial_advanced_packaging.html?highlight=setup_environment#set-environment-variables-in-your-own-package) for the doc and eg [here](https://github.com/spack/spack/blob/54f97d1dec8df18b3d34144f38760abd9a7cc5b4/var/spack/repos/builtin/packages/py-cvxopt/package.py) for example usage

Reply:
Check